src/aurora/ltrfile.cpp

```diff
@@ -0,0 +1,186 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  File for creating random names in the character generation.
+ */
+
+#include "src/common/strutil.h"
+
+#include "src/aurora/ltrfile.h"
+#include "src/aurora/resman.h"
+
+static const uint32 kLTRID     = MKTAG('L', 'T', 'R', ' ');
+static const uint32 kVersion10 = MKTAG('V', '1', '.', '0');
+
+// TODO Check for non-latin language alphabets
+static const char *kLetters28 = "abcdefghijklmnopqrstuvwxyz'-";
+static const char *kLetters26 = "abcdefghijklmnopqrstuvwxyz";
+
+namespace Aurora {
+
+LTRFile::LTRFile(const Common::UString &ltr) {
+	Common::ScopedPtr<Common::SeekableReadStream> stream(ResMan.getResource(ltr, kFileTypeLTR));
+	load(stream.get());
+}
+
+LTRFile::LTRFile(Common::SeekableReadStream *stream) {
```

Comment:
LTRFile doesn't use the stream after loading anymore, right? So it should take a `Common::SeekableReadStream &stream`, to indicate that the LTRFile doesn't take over the stream.